app/assets/javascripts/Components/__tests__/marks_panel.test.jsx

```diff
@@ -0,0 +1,425 @@
+  const getWrapper = props => {
+    return shallow(<MarksPanel {...props} />);
+  };
+  beforeEach(() => {
+    basicProps = {
+      max_mark: 100,
+      released_to_students: false,
+      old_marks: {},
+      marks: [],
+      levels: [],
+      assigned_criteria: [],
+      annotations: [],
+      updateMark: jest.fn(),
+      revertToAutomaticDeductions: jest.fn(),
+      findDeductiveAnnotation: jest.fn(),
+    };
+  });
+  it("displays no criterion if marks is an empty list", () => {
+    const wrapper = getWrapper();
+    expect(wrapper.find("marks-list").length).toBe(0);
+  });
+  it("displays a criterion of the right type when given an array of marks", () => {
+    basicProps.marks.push({
+      criterion_type: "CheckboxCriterion",
+      id: 67,
+      mark: 0,
+      description: "testing",
+    });
+    basicProps.old_marks[67] = [];
+    basicProps.marks.push({
+      criterion_type: "FlexibleCriterion",
+      id: 66,
+      mark: 0,
+      description: "testing",
+    });
+    basicProps.old_marks[66] = [];
+    basicProps.marks.push({
+      criterion_type: "RubricCriterion",
+      id: 65,
+      mark: 0,
+      description: "testing",
+    });
+    basicProps.old_marks[65] = [];
+    const wrapper = getWrapper(basicProps);
+    basicProps.marks.forEach(mark => {
+      expect(wrapper.find(`#${convertToKebabCase[mark.criterion_type]}_${mark.id}`)).toBeTruthy();
+    });
+  });
+});
+
+describe("CheckboxCriterionInput", () => {
+  let wrapper, basicProps;
+  beforeEach(() => {
+    basicProps = {
+      id: 67,
+      mark: 0,
+      max_mark: 1,
+      expanded: false,
+      unassigned: false,
+      released_to_students: false,
+      oldMark: {},
+      description: " ",
+      updateMark: jest.fn(),
+      destroyMark: jest.fn(),
+      toggleExpanded: jest.fn(),
+    };
+    wrapper = shallow(<CheckboxCriterionInput {...basicProps} />);
+  });
+  it("should toggle expand and contract upon clicking the expand/contract button", () => {
+    basicProps.toggleExpanded.mockImplementation(() => {
+      basicProps.expanded = !basicProps.expanded;
+    });
+    wrapper.find(`#checkbox_criterion_${basicProps.id}_expand`).simulate("click");
+    expect(basicProps.toggleExpanded).toHaveBeenCalled();
+    expect(basicProps.expanded).toBeTruthy();
+
+    wrapper.find(`#checkbox_criterion_${basicProps.id}_expand`).simulate("click");
+    expect(basicProps.toggleExpanded).toHaveBeenCalled();
+    expect(basicProps.expanded).toBeFalsy();
+  });
+  it("it delete upon the delete button being pressed", () => {
+    basicProps.unassigned = false;
+    basicProps.mark = 1;
+    wrapper.setProps(basicProps);
+
+    wrapper.find(`#checkbox_criterion_${basicProps.id}_destroy`).simulate("click");
+    expect(basicProps.destroyMark).toHaveBeenCalledWith(undefined, basicProps.id);
+  });
+  it("correctly updates mark to max_mark and 0 when clicking on respective buttons", () => {
+    basicProps.updateMark.mockImplementation((id, new_mark) => {
```

Comment:
same comment as above

Reply:
Same comment as above for me (it is only mocked once per describe)